vms/platformvm/service.go

```diff
@@ -1200,6 +1210,102 @@ func (service *Service) AddDelegator(_ *http.Request, args *AddDelegatorArgs, re
 	return errs.Err
 }
 
+// AddLockArgs are the arguments to AddLock
+type AddLockArgs struct {
+	// User, password, from addrs, change addr
+	api.JSONSpendHeader
+	APIStaker
+	RewardAddress string `json:"rewardAddress"`
+}
+
+// AddLock creates and signs and issues a transaction to lock tockens
+func (service *Service) AddLock(_ *http.Request, args *AddLockArgs, reply *api.JSONTxIDChangeAddr) error {
```

Comment:
This for sure needs test as well

Reply:
Yep, test is TODO matter, but I heard a lot of talks that they'd be outsourced, so I'm waiting for clarification on that topic